### DISCUSSION TITLE
Getting server Id from inside a running instance

### DISCUSSION THREAD

**Discussion by C0D3-M4513R** [pterodactyl/panel#4615] [Support] [answered]
I am developing a plugin targeting Minecraft servers, which will send a restart and/or kill signal to perodactyl, should the server not tick for a specified amount of seconds.
My issue is, that I need the server's identifier/uuid to send the reboot/kill signals to the correct signal to the correct server. 
Using java to get the current pid and then killing it is also not really possible, because we still use minecraft 1.12 and are thus confined to jdk8 (which has no support for getting the current pid, without some reflection hacks that don't work for us, because the java security handler disallows it). Also I'm not sure if perodactyl would notice that the server is dead if another pocess kills it.
Similarly I could try and do some thing with os commands, such as `ps -aux | grep ${java-commandline here}`, but that might also not be 100% reliable.

Is there any way to get the current server's uuid, from inside the container without modifying files outside the container? or can support for this be added? 
Otherwise we could see which information can be queried from inside the docker container (such as the docker ip), which could identify the server.

**Comment by parkervcp** [comment]
The UUID is a variable that is passed into the container. You can use that with a get on servers to get the server id itself.

I am pretty sure it's the variable `P_UUID` but I would need to check.

**Reply by C0D3-M4513R** [reply]
Starting a jar from the following code outputs:
`UUID is 'null'.`
```java
package org.example;

public class Main {
    public static void main(String[] args) {

        System.out.println("UUID is '"+System.getenv("P_UUID")+"'.");
    }
}
```

**Reply by C0D3-M4513R** [reply]
It may be possible, that it is `P_SERVER_UUID`?

**Comment by trenutoo** [comment, marked as answer]
All default variables are [documented here.](https://pterodactyl.io/community/config/eggs/creating_a_custom_egg.html#list-of-default-variables)

It is `P_SERVER_UUID` as [seen in the source code.](https://github.com/pterodactyl/panel/blob/1.0-develop/app/Services/Servers/EnvironmentService.php#L70)
